src/sql/platform/connection/common/constants.ts

```diff
@@ -19,6 +19,7 @@ export const connectionProviderContextKey = 'connectionProvider';
 export const applicationName = 'azdata';
 
 export const defaultEngine = 'defaultEngine';
+export const sqlOnDemand: number = 11;
```

Comment:
If you're going to need to define this somewhere it should be as an enum for all editions - not a special case number like this

Reply:
I haven't seen the enum for engine editions anywhere.
 Do you suggest creating new enum for all editions? Would this file be appropriate place for it?

 I've added this constant to avoid having magical numbers in the code base.

Reply:
Yeah I'm surprised we don't have one yet. It'd be great if you added an enum to azdata.d.ts for this, I'd suggest putting it in the root namespace (where ServerInfo is defined)
```
    export enum DatabaseEngineEdition
    {
        //
        Unknown = 0,
        Personal = 1,
        Standard = 2,
        Enterprise = 3,
        Express = 4,
        SqlDatabase = 5,
        SqlDataWarehouse = 6,
        SqlStretchDatabase = 7,
        SqlManagedInstance = 8,
        SqlOnDemand = 11
    }

---
In reply to: [335550074](https://github.com/microsoft/azuredatastudio/pull/7695#discussion_r335550074) [](ancestors = 335550074)